app/lib/request.rb

```diff
@@ -148,6 +149,24 @@ def http_client
 
   private
 
+  def normalize_preserving_url_encodings(url, preserve_chars = SAFE_PRESERVED_CHARS)
+    original_uri = Addressable::URI.parse(url)
+
+    if original_uri.query
+      normalized_query = Addressable::URI.normalize_component(
```

Comment:
I'm looking at the original implementation and it's pretty different:
https://github.com/sporkmonger/addressable/blob/3450895887d0a1770660d8831d1b6fcfed9bd9d6/lib/addressable/uri.rb#L1609-L1635

I'm not sure how far from that we want to go.